static/app/views/performance/trends/changeExplorer.tsx

```diff
@@ -0,0 +1,330 @@
+import {t, tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Organization, Project} from 'sentry/types';
+import theme from 'sentry/utils/theme';
+import useKeyPress from 'sentry/utils/useKeyPress';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+import {MetricsChart} from 'sentry/views/performance/trends/changeExplorerUtils/metricsChart';
+import {Chart} from 'sentry/views/performance/trends/chart';
+import {
+  NormalizedTrendsTransaction,
+  TrendChangeType,
+  TrendParameter,
+  TrendsStats,
+  TrendView,
+} from 'sentry/views/performance/trends/types';
+import SlideOverPanel from 'sentry/views/starfish/components/slideOverPanel';
+
+type PerformanceChangeExplorerProps = {
+  collapsed: boolean;
+  isLoading: boolean;
+  location: Location;
+  onClose: () => void;
+  organization: Organization;
+  projects: Project[];
+  statsData: TrendsStats;
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+  trendFunction: string;
+  trendParameter: TrendParameter;
+  trendView: TrendView;
+};
+
+type ExplorerBodyProps = {
+  isLoading: boolean;
+  location: Location;
+  organization: Organization;
+  projects: Project[];
+  statsData: TrendsStats;
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+  trendFunction: string;
+  trendParameter: TrendParameter;
+  trendView: TrendView;
+};
+
+type HeaderProps = {
+  transaction: NormalizedTrendsTransaction;
+  trendChangeType: TrendChangeType;
+};
+
+export function PerformanceChangeExplorer({
+  collapsed,
+  transaction,
+  onClose,
+  trendChangeType,
+  trendFunction,
+  trendView,
+  statsData,
+  isLoading,
+  organization,
+  projects,
+  trendParameter,
+  location,
+}: PerformanceChangeExplorerProps) {
+  const panelRef = useRef<HTMLDivElement>(null);
+  useOnClickOutside(panelRef, () => {
+    if (!collapsed) {
+      onClose();
+    }
+  });
+
+  const escapeKeyPressed = useKeyPress('Escape');
+
+  useEffect(() => {
+    if (escapeKeyPressed) {
+      if (!collapsed) {
+        onClose();
+      }
+    }
+  }, [escapeKeyPressed, collapsed, onClose]);
+
+  return (
+    <SlideOverPanel collapsed={collapsed} ref={panelRef}>
+      <CloseButtonWrapper>
+        <CloseButton
+          priority="link"
+          size="zero"
+          borderless
+          aria-label={t('Close Details')}
+          icon={<IconClose size="sm" />}
+          onClick={onClose}
+        />
+      </CloseButtonWrapper>
+      <PanelBodyWrapper>
+        <ExplorerBody
+          transaction={transaction}
+          trendChangeType={trendChangeType}
+          trendFunction={trendFunction}
+          trendView={trendView}
+          statsData={statsData}
+          isLoading={isLoading}
+          organization={organization}
+          projects={projects}
+          trendParameter={trendParameter}
+          location={location}
+        />
+      </PanelBodyWrapper>
+    </SlideOverPanel>
```

Comment:
Could we be using `DetailPanel` (which already includes the hooks for handling collapse)? 

There is a concept that some components are 'controlled' or not (whether they manage their own state), in this case you want an 'uncontrolled' `DetailPanel` so you can pass in collapsed. I think you are only using `collapse` as `true`, the `DetailPanel` handles this by creating/destroying the drawer instead (when you render it it appears automatically). The `DetailPanel` should likely be switched over to allowed for opt-out w/ controlled, but for now you can probably use it the same way.